File: product-frontend/src/services/productService.js
import api from "./axiosConfig";

const API_URL = "/api/products";

export const getProducts = (page, size, sortBy, direction) => {
  return api.get(
    `${API_URL}?page=${page}&size=${size}&sortBy=${sortBy}&direction=${direction}`
  );
};

export const createProduct = (product) => {
  return api.post(API_URL, product);
};

export const getProductById = (id) => {
  return api.get(`${API_URL}/${id}`);
};

export const updateProduct = (id, product) => {
  return api.put(`${API_URL}/${id}`, product);
};

export const deleteProduct = (id) => {
  return api.delete(`${API_URL}/${id}`);
};

export const filterProducts = (minPrice, maxPrice) => {
  return api.get(
    `${API_URL}/filter?minPrice=${minPrice}&maxPrice=${maxPrice}`
  );
};

export const searchProducts = (name) => {
  return api.get(`${API_URL}/search?name=${name}`);
};

File: productapi/src/main/java/com/suresh/productapi/service/ProductService.java
package com.suresh.productapi.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.suresh.productapi.dto.ProductDTO;
import com.suresh.productapi.dto.ProductResponseDTO;

public interface ProductService {

    ProductResponseDTO createProduct(ProductDTO dto);

    ProductResponseDTO getProductById(Long id);

    Page<ProductResponseDTO> getAllProducts(int page, int size, String sortBy, String direction);

    List<ProductResponseDTO> filterProductsByPrice(double minPrice, double maxPrice);

    List<ProductResponseDTO> searchProducts(String name);

    ProductResponseDTO updateProduct(Long id, ProductDTO dto);

    void deleteProduct(Long id);
}